addons/hr_payroll/models/hr_contract.py

```diff
@@ -0,0 +1,36 @@
+#-*- coding:utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class HrContract(models.Model):
+    """
+    Employee contract based on the visa, work permits
+    allows to configure different Salary structure
+    """
+
+    _inherit = 'hr.contract'
+    _description = 'Employee Contract'
+
+    struct_id = fields.Many2one('hr.payroll.structure', 'Salary Structure')
+    schedule_pay = fields.Selection([
+        ('monthly', 'Monthly'),
+        ('quarterly', 'Quarterly'),
+        ('semi-annually', 'Semi-annually'),
+        ('annually', 'Annually'),
+        ('weekly', 'Weekly'),
+        ('bi-weekly', 'Bi-weekly'),
+        ('bi-monthly', 'Bi-monthly'),
+    ], 'Scheduled Pay', index=True, default='monthly')
+
+    @api.multi
+    def get_all_structures(self, contract_ids):
+        """
+        @param contract_ids: list of contracts
+        @return: the structures linked to the given contracts, ordered by hierachy (parent=False first, then first level children and so on) and without duplicate
+        """
+        structure_ids = [contract.struct_id.id for contract in self.browse(contract_ids) if contract.struct_id]
```

Comment:
now use simply if self contain contract recordset then 
self.mapped('struct_id') which return structure recordset and then call  _get_parent_structure() with this recordset.